sampleExtension/background.js

```diff
@@ -0,0 +1,121 @@
+
+//Set that holds the urls to be intercepted
+var blockedSites = [];
+
+initExtension = function () {
+    //First receive the blacklist from the sync storage, and then create a onBeforeRequest listener using this list.
+    updateBlockedSites(replaceListener);
+    addBrowserActionListener();
+};
+
+// This function receives the blacklist from the sync storage.
+updateBlockedSites = function(callback){
+    chrome.storage.sync.get("tds_blacklist", function (items) {
+        if (!chrome.runtime.error) {
+            blockedSites = items.tds_blacklist;
+            return callback();
+        }
+    });
+};
+
+// This function adds one url to the blacklist
+addToBlockedSites = function(urlToAdd) {
+    var formatUrl = urlToAdd.split(['//'])[1];
```

Comment:
Different levels of abstraction in one function, introduce auxilary function, e.g. `formatTheUrl()`.